plugins/inputs/tail/tail.go

```diff
@@ -179,43 +183,35 @@ func (t *Tail) receiver(parser parsers.Parser, tailer *tail.Tail) {
 			}
 			firstLine = false
 		} else {
-			m, err = parser.ParseLine(text)
+			m, err = parser.ParseLine(line.Line)
 		}
 
 		if err == nil {
 			if m != nil {
 				tags := m.Tags()
-				tags["path"] = tailer.Filename
+				tags["path"] = line.Filename
 				t.acc.AddFields(m.Name(), m.Fields(), tags, m.Time())
 			}
 		} else {
 			t.acc.AddError(fmt.Errorf("E! Malformed log line in %s: [%s], Error: %s\n",
-				tailer.Filename, line.Text, err))
+				line.Filename, line.Line, err))
 		}
 	}
-
-	log.Printf("D! [inputs.tail] tail removed for file: %v", tailer.Filename)
-
-	if err := tailer.Err(); err != nil {
-		t.acc.AddError(fmt.Errorf("E! Error tailing file %s, Error: %s\n",
-			tailer.Filename, err))
-	}
 }
 
 func (t *Tail) Stop() {
 	t.Lock()
 	defer t.Unlock()
 
-	for _, tailer := range t.tailers {
-		err := tailer.Stop()
-		if err != nil {
-			t.acc.AddError(fmt.Errorf("E! Error stopping tail on file %s\n", tailer.Filename))
+	err := t.tailer.Close()
```

Comment:
Can you describe what, if anything, on tailer can be used concurrently?  Maybe we don't need the locks anymore.  Telegraf will never call Start(), Gather(), Stop() at the same time, so we only need to worry about Stop and any processing goroutines.